crates/hotshot/src/tasks/mod.rs

```diff
@@ -87,12 +92,15 @@ pub async fn add_network_message_task<TYPES: NodeType, NET: CommunicationChannel
         let network = net.clone();
         let closure = async move {
             loop {
-                let msgs = Messages(
-                    network
-                        .recv_msgs(TransmitType::Direct)
-                        .await
-                        .expect("Failed to receive direct messages"),
-                );
+                let msgs = match network.recv_msgs(TransmitType::Direct).await {
+                    Ok(msgs) => Messages(msgs),
+                    Err(err) => {
+                        error!("failed to receive broadcast messages: {err}");
```

Comment:
nit: broadcast should be direct

Also same question, do we hit this case sometimes?